CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs

```diff
@@ -308,62 +307,93 @@ private static void WriteNullProperty(string propertyAssignment, CodeProperty pa
+        CodeProperty parentProperty, CodeProperty codeProperty, IndentManager indentManager, bool fromMap = false)
     {
         var hasSchema = codeProperty.PropertyType == PropertyType.Object;
         var arrayName = $"{objectName.ToFirstCharacterLowerCase()}Array";
         StringBuilder builder = new StringBuilder();
         if (hasSchema) 
             builder.AppendLine($"${arrayName} = [];");
+        else if (fromMap)
+        {
+            builder.AppendLine($"{indentManager.GetIndent()}[");
+            indentManager.Indent();
+        }
         else if (codeProperty.Children.FirstOrDefault().PropertyType != PropertyType.Object)
-             builder.Append('[');
+        {
+            builder.Append($"{indentManager.GetIndent()}[");
+            indentManager.Indent();
+        }
+
         int childPosition = 0;
         CodeProperty lastProperty = default;
         foreach (var property in codeProperty.Children)
         {
             var childPropertyName = $"{EscapePropertyNameForSetterAndGetter(codeProperty.Name)}{EscapePropertyNameForSetterAndGetter(property.Name)}{++childPosition}".ToFirstCharacterLowerCase();
-            WriteCodeProperty(propertyAssignment, builder, codeProperty, property, indentManager, childPosition, childPropertyName);
-            if (property.PropertyType == PropertyType.Object && codeProperty.PropertyType == PropertyType.Array)
+            var propertyCopy = property;
+            if (fromMap) propertyCopy.PropertyType = PropertyType.Map;
+
+            WriteCodeProperty(propertyAssignment, builder, codeProperty, propertyCopy, indentManager, childPosition,
+                    childPropertyName, fromMap: fromMap);
+
+            if (property.PropertyType == PropertyType.Object && codeProperty.PropertyType == PropertyType.Array && !fromMap)
             {
                 builder.AppendLine($"${arrayName} []= ${childPropertyName};");
             }
 
             lastProperty = property;
         }
 
-        if (lastProperty.PropertyType == PropertyType.Object && codeProperty.PropertyType == PropertyType.Array)
+        if (lastProperty.PropertyType == PropertyType.Object && codeProperty.PropertyType == PropertyType.Array && !fromMap)
         {
             builder.AppendLine(
                 $"${propertyAssignment}->set{EscapePropertyNameForSetterAndGetter(codeProperty.Name)}(${arrayName});");
             payloadSb.AppendLine(builder.ToString());
         }
         else
         {
-            builder.Append(']');
+            builder.Append($"{indentManager.GetIndent()}]");
             if (parentProperty.PropertyType == PropertyType.Object)
                 payloadSb.AppendLine(
-                    $"${propertyAssignment}->set{EscapePropertyNameForSetterAndGetter(codeProperty.Name)}({builder.ToString()});");
+                    $"${propertyAssignment}->set{EscapePropertyNameForSetterAndGetter(codeProperty.Name)}({builder.ToString().Trim(' ')});");
```

Comment:
```suggestion
                    $"${propertyAssignment}->set{EscapePropertyNameForSetterAndGetter(codeProperty.Name)}({builder.ToString().Trim()});");
```
I suspect you want to remove whitespace not space characters. Otherwise, the extra spaces added by `Indent()` will persist as they are tab characters not space characters.